### PR TITLE
Add generated .gitignore.

### PR DESCRIPTION
**Pull request by Math-ias**
The .gitignore was generated with gitignore.io with the tags:
 * Linux
 * OSX
 * Windows
 * Java
 * Vim
The gitignore does not yet ignore the directory in which javadocs are contained.
This has proven to be useful as javadocs are a big clunky set of files that are
expensive to store with git. And by committing changes to javadocs graphs are
screwed up on GitHub.

### CODE REVIEW

**Comment by Math-ias** [comment]
@TheLocust3 can you approve this or find someone to do it?

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+*.ear
+
+# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
+hs_err_pid*
+
+
+### Eclipse ###
+*.pydevproject
+.metadata
+.gradle
```
Comment (TheLocust3) [inline .gitignore]: Maybe .gradle shouldn't be under a different category because it is not eclipse specific.

**Comment by TheLocust3** [comment]
Overall the items under eclipse are not Eclipse specific. In fact most of them have nothing to do with Eclipse like .gradle and .nib. Consider renaming this or moving some items out of that list.

**Comment by Math-ias** [comment]
I didn't add those myself, they were generated, will fix however.

**Comment by Math-ias** [comment]
Updated @TheLocust3 ready to merge.

**Comment by TheLocust3** [comment]
Mathias assignee me not tag me

**Comment by Math-ias** [comment]
Sorry about the tagging. I have recategorized .swp and .nib.

**Comment by TheLocust3** [comment]
+1 (That means you can merge it)

**Comment by Math-ias** [comment]
So that's a go on merging?
I will also probably cherry-pick these in our run-queue-client repository.

**Comment by TheLocust3** [comment]
Go ahead and merge it. You should also add this gitignore to the client repo (no pull request needed).

**Pull request merged**